library/built-in/_prefabs/ControlNet/prefab_cnet_ipAdapter.ts

```diff
@@ -79,11 +79,13 @@ export const ui_subform_IPAdapter = () => {
 }
 
 // 🅿️ IPAdapter RUN ===================================================
-export const run_cnet_IPAdapter = async (
+export const run_cnet_IPAdapter = (
     IPAdapter: OutputFor<typeof ui_subform_IPAdapter>,
     cnet_args: Cnet_args,
     image: _IMAGE,
-) => {
+): {
+    ip_adapted_model: _MODEL
+} => {
```

Comment:
I tend to like having return types explicitly added; 
makes typescript not slow down when project grows, and allow for various errors to appear at better locations
also, simpler to read.